GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -65,6 +68,28 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
             generateNotifications();
         } else if ("purgeDeviceFileJobQueueRecords".equals(action)) {
             purgeDeviceFileJobQueueRecords();
+        } else if ("purgeReportRecords".equals(action)) {
+            purgeReportRecords();
+        }
+    }
+
+    /**
+     * scans for and deletes Report entries that are either more than one year old
+     */
+    private void purgeReportRecords() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, ONE_YEAR_AGO);
+        log.fine("Starting scan for Report entries, older than: " + deadline.getTime());
+        ReportDao reportDao = new ReportDao();
+        List<Report> reportList = reportDao.listAllCreatedBefore(deadline.getTime());
+        log.fine("Attempting to retire " + reportList.size());
+        for (Report item : reportList) {
+        	log.fine("Deleting old Report entry: " + item.getKey().getId());
+        	//TODO delete file from file store
+        	//If that succeeded:
+        	SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_REPORT_ACTION,
```

Comment:
Were you able to look into a batch delete function?

Reply:
Are you sure we always have time for that in the foreground?

Reply:
Cron jobs run in the background and do not run into the same limit of one-minute to complete a task so it should be ok.